### PR TITLE
Boundaries object oriented

### LINKED ISSUE #3134: Boundaries object oriented
### Description of the feature

Right now boundaries do not rely on AEDT object-oriented, it has some limitations, if you add a boundary manually or a 3DComponent, the pyaedt object will no be updated and you will need to open and close the project again to find the new boundaries.

### Steps for implementing the feature

_No response_

### Useful links and references

_No response_

### PR DESCRIPTION
**Pull request by Samuelopez-ansys**
Close #3134

### CODE REVIEW

File: pyaedt/application/Design.py [hunk 1]
```diff
@@ -297,9 +297,140 @@ def boundaries(self):
...
+                _dict_out[bound.type] = [bound]
+        return _dict_out
+
+    @property
+    def design_excitations(self):
+        """Design excitations.
+
+        Returns
+        -------
+        List of :class:`pyaedt.modules.Boundary.BoundaryObject`
```
Comment (PipKat) [inline pyaedt/application/Design.py:402]:
```suggestion
        list
            List of :class:`pyaedt.modules.Boundary.BoundaryObject`.
```

File: pyaedt/icepak.py [hunk 2]
```diff
@@ -3930,21 +3932,18 @@ def assign_solid_block(
             ``"Joule Heating"``. For a temperature-dependent or transient
             assignment, a dictionary can be used. The dictionary should contain three keys:
             ``"Type"``, ``"Function"``, and ``"Values"``.
-
```
Comment (PipKat) [inline pyaedt/icepak.py:3933]: Removing the blank line means that the bulleted list will not render properly in the documentation. You MUST have a blank line before the first item in the list.

File: pyaedt/modules/Boundary.py [hunk 3]
```diff
@@ -296,9 +346,9 @@ class BoundaryObject(BoundaryCommon, object):
         An AEDT application from ``pyaedt.application``.
```
Comment (PipKat) [inline pyaedt/modules/Boundary.py:346]: The Google developer documentation guidelines say that a literal string should be used as an adjective, with a noun afterwards. Is ``pyaedt.application`` a module, a class, or what? Also, we generally omit the "A, An, or The" in the first summary line of a description.

File: pyaedt/modules/Boundary.py [hunk 4]
```diff
@@ -296,9 +346,9 @@ class BoundaryObject(BoundaryCommon, object):
...
     name : str
         Name of the boundary.
-    props : dict
+    props : dict, optional
```
Comment (PipKat) [inline pyaedt/modules/Boundary.py:349]: What is the default? What is the behavior for the default?

File: pyaedt/modules/Boundary.py [hunk 5]
```diff
@@ -315,15 +365,82 @@ class BoundaryObject(BoundaryCommon, object):
...
+                        child_object = self._app.get_oo_object(self._app.odesign, "Conductors\\{}".format(port))
+
+        if child_object:
+            return BinaryTreeNode(self.name, child_object, False)
+
+        return False
+
+    @property
+    def props(self):
+        """Retrieve boundary data.
```
Comment (PipKat) [inline pyaedt/modules/Boundary.py:431]:
```suggestion
        """Boundary data.
```

File: pyaedt/modules/Boundary.py [hunk 6]
```diff
@@ -333,6 +450,15 @@ def type(self):
         str
             Returns Type of the boundary
```
Comment (PipKat) [inline pyaedt/modules/Boundary.py:451]:
```suggestion
            Returns the type of the boundary.
```

File: pyaedt/modules/Boundary.py [hunk 7]
```diff
@@ -744,15 +875,54 @@ class MaxwellParameters(BoundaryCommon, object):
...
+            child_object = self._app.odesign.GetChildObject("Parameters").GetChildObject(self.name)
+        elif self.name in self._app.odesign.GetChildObject("Parameters").GetChildNames():
+            child_object = self._app.odesign.GetChildObject("Parameters").GetChildObject(self.name)
+        if child_object:
+            return BinaryTreeNode(self.name, child_object, False)
+        return False
+
+    @property
+    def props(self):
+        """Retrieve maxwell parameters data.
```
Comment (PipKat) [inline pyaedt/modules/Boundary.py:913]:
```suggestion
        """Maxwell parameter data.
```

File: pyaedt/modules/Boundary.py [hunk 8]
```diff
@@ -744,15 +875,54 @@ class MaxwellParameters(BoundaryCommon, object):
...
+        props = self._get_boundary_data()
+
+        if self.name in props:
+            self._props = BoundaryProps(self, OrderedDict(props[self.name][0]))
+            self._type = props[self.name][1]
+        return self._props
+
     @property
     def name(self):
         """Boundary Name."""
```
Comment (PipKat) [inline pyaedt/modules/Boundary.py:928]:
```suggestion
        """Boundary name."""
```

File: pyaedt/modules/Boundary.py [hunk 9]
```diff
@@ -1635,6 +1828,21 @@ def name(self, value):
         self.update()
         self._name = value
 
+    @property
+    def props(self):
+        """Retrieve excitation data.
```
Comment (PipKat) [inline pyaedt/modules/Boundary.py:1833]:
```suggestion
        """Excitation data.
```

File: pyaedt/icepak.py [hunk 10]
```diff
@@ -3930,21 +3932,18 @@ def assign_solid_block(
...
+            set to ``"Transient"``, acceptable values are `"Exponential"``, `"Linear"``,
+            ``"Piecewise Linear"``, ``"Power Law"``, ``"Sinusoidal"``, and ``"SquareWave"``.
             - For the ``"Values"`` key, a list of strings contain the parameters required by
-              the ``"Function"`` key selection. For example, whn``"Linear"`` is set as the
-              ``"Function"`` key, two parameters are required: the value of the variable
-              at t=0 and the slope of the line. For the parameters required by each
-              ``"Function"`` key selection, see the Icepak documentation. The parameters
-              must contain the units where needed.
-
+            the ``"Function"`` key selection. For example, whn``"Linear"`` is set as the
```
Comment (MaxJPRey) [inline pyaedt/icepak.py:3942]:
```suggestion
            the ``"Function"`` key selection. For example, when``"Linear"`` is set as the
```

File: pyaedt/modules/Boundary.py (hunk 3 above)
Comment (Samuelopez-ansys) [reply pyaedt/modules/Boundary.py:346]: @PipKat It is a class, how should I modify it? 

For example?

app : :class:`pyaedt.application`
        Pyaedt object from ``pyaedt.application``.

**Review by maxcapodi78 — APPROVED** [review]
Great Enhancement

**Pull request merged**